src/java/main/org/apache/zookeeper/server/auth/ProviderRegistry.java

```diff
@@ -33,7 +33,7 @@
     private static HashMap<String, AuthenticationProvider> authenticationProviders =
         new HashMap<String, AuthenticationProvider>();
 
-    public static void initialize() {
+    public static void initialize(ZooKeeperServer zks) {
```

Comment:
I don't quite like the idea of passing a `ZooKeeperServer` instance here, it is too specific. See suggestion below.